docs/Controllers.md

```diff
@@ -0,0 +1,120 @@
+
+## Client Side Implementation Notes
+
+Controllers MUST adhere the Client Side Implementation Notes described in this specification, namely the sections on [Conformance to Schemas](APIs%20-%20Client%20Side%20Implementation.md#conforming-to-schemas), [RTP Operating Point](APIs%20-%20Client%20Side%20Implementation.md#rtp-operating-point), and [Failure Modes](APIs%20-%20Client%20Side%20Implementation.md#failure-modes).
+
+## Interoperability
+
+### Identifying Active Connections
+
+As is described in the [Identifying Active Connections section of the Interoperability: IS-04 document](Interoperability%20-%20IS-04.md#identifying-active-connections) in this specification,
+in order to populate the subscription attribute of IS-04 Senders and Receivers, the Connection API includes keys for `sender_id` and `receiver_id` in its `/staged` parameters.
+These are used to signal that a Sender or Receiver is connected to an NMOS Receiver or Sender. 
+The Controller MUST set and unset (using `null`) the `sender_id` or `receiver_id` parameters when modifying the `transport_params` or `transport_file`.
+
+### Interacting with Non-NMOS Devices
+
+Controllers MUST follow the requirements for a Client interacting with non-NMOS Devices, as described in the [Interoperability: Non-NMOS Devices](Interoperability%20-%20Non-NMOS%20Devices.md) section of this Specification.
+
+## Behaviour
+
+### Operation with SMPTE 2022-7
+
+When the Controller is interacting with Receivers that support SMPTE 2022-7 it MUST follow the requirements for Clients described in the [Operation with SMPTE 2022-7 section of the Behaviour - RTP Transport Type document](Behaviour%20-%20RTP%20Transport%20Type.md#operation-with-smpte-2022-7) in this specification.
+
+### Version Timestamp
+
+The Controller SHOULD use an IS-04 Query API WebSocket connection to monitor version timestamp increments on Senders and Receivers being controlled.
+
+A version timestamp increment indicates that another Controller has possibly changed resource parameters, and as such any information cached by the Controller could possibly be stale.
+
+### Sender Multicast Address
+
+The Controller SHOULD allow allocation of multicast addresses to allow avoidance of conflicts.
+To avoid conflicts, when a Sender is added to a new system, the Controller MUST allow the use of IS-05 to configure the multicast transmit information (`transport_params`) before or in the same transaction as enabling the transmitter (`master_enable` = `true`).
+
+### TCP Sessions
+
+The Controller SHOULD use persistent TCP connections wherever possible,
+to allow multiple operations to be performed without repeatedly closing and reopening connections for each operation.
+
+The use of HTTP pipelining has been known to cause interoperability issues in some cases.
+
+If the Controller uses HTTP pipelining, it SHOULD follow the guidelines set out in [RFC 7230 section 6.3.2](https://datatracker.ietf.org/doc/html/rfc7230#section-6.3.2).
```

Comment:
I think since RFC 7230 section 6.3.2 itself uses conformance language, this statement needs either to say MUST or be written as fact, e.g.

```suggestion
If the Controller uses HTTP pipelining, the guidelines set out in [RFC 7230 section 6.3.2](https://datatracker.ietf.org/doc/html/rfc7230#section-6.3.2) apply.
```

Reply:
Should this document also say something about the use of concurrent connections? See e.g. https://datatracker.ietf.org/doc/html/rfc7230#section-6.4. Note that some implementations of NMOS APIs have been known which do not support concurrent connections...

Reply:
ARG - accept suggestion but the RFC section on concurrent connections is next to useless so no point referencing.